File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    box = []
    answer = []
    for i in arr:
        if i in box or len(box) == 0:
            box.append(i)
        else:
            a = box.pop()
            answer.append(a)
            box = []
            box.append(i)
    a = box.pop()
    answer.append(a)
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
    List <Integer> waiting = new ArrayList<>();
    List <Integer> answer = new ArrayList<>();
    for (int i : arr){
        int last_id = waiting.size()-1;
        if (waiting.isEmpty()){
            waiting.add(i);
        }

        else if (waiting.get(last_id) != i){
            answer.add(waiting.get(last_id));
            waiting.remove(last_id);
            waiting.add(i);
        }
    }

    if (waiting.size() > 0){ // 남은 거 처리
        answer.add(waiting.get(waiting.size()-1));
    }

    return answer.stream().mapToInt(i -> i).toArray();
}
}